src/renderer/components/ContextMenu.tsx

```diff
@@ -182,12 +219,25 @@ export function ContextMenu(props: {
         } else {
           ;(parent?.lastElementChild as HTMLDivElement).focus()
         }
+      } else if (ev.key == 'ArrowLeft') {
+        setSublevels(l => l.slice(0, Math.max(0, l.length - 1)))
+        keyboardFocus.current = openSublevels[openSublevels.length - 1]
+      } else if (ev.key == 'ArrowRight') {
+        if (current) {
+          const index = +((current as HTMLDivElement).dataset
+            .expandableIndex as string)
+          if (!isNaN(index)) {
+            setSublevels(l => [...l, index])
+            keyboardFocus.current = 0
+          }
+        }
```

Comment:
I find this code a little hard to read. Some suggestions for a refactoring:

1. Every time you change the value of `sublevels` some slicing, spreading etc. is used which are hard to read. Why not move these `setSublevels` into separate methods and give them an explicit names what they are doing? For example:
    ```js
    const expandSubMenu = () => {
      // ...
    }
    
    const collapseSubMenu = () => {
      // ...
    }
    ``` 
    This approach might also fit nicely into the suggestion below to move the `onClick` etc. event handler logic into explicit methods (for example `closeMenu`) as well. They could all live next to each other and show how we can interact with this "sublevel" data structure
2. Type annotations and code in the same line can be hard to read, I'd separate things like this into separate lines, otherwise one almost misses the actual crucial bit which is that this type got converted by the `+`, which I would also make explicit
    ```js
    const index = +((current as HTMLDivElement).dataset
      .expandableIndex as string)
    
    // vs.
    
    const current = parent?.querySelector(':focus') as HTMLDivElement
    const index = parseInt(current.dataset.expandableIndex, 10)
    
    // .. or similar
    ```

Reply:
I like the parseInt solution more, it is more clear than the `+` which is close to ninja code